src/presentations/Cart.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import LazyLoad from 'react-lazyload';
+import {resetCartData} from '../actions/productActions';
+class Cart extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            itemList: [],
+            totalCartValue:0
+        }
+        // this.increaseQty = this.increaseQty.bind(this);
+        // this.decreaseQty = this.decreaseQty.bind(this);
+        // this.evaluateTotalPrice = this.evaluateTotalPrice.bind(this);
+    }
+    componentDidMount(){
+        if(this.props.cartData){
+            this.setState({itemList:this.props.cartData},()=>{
+                this.evaluateTotalPrice();
+            });
+        }
+    }
+    increaseQty=(e,itm,i)=>{
+        let tempObj = Object.assign({},this.state);
+        itm.qty = itm.qty + 1;
+        tempObj.itemList[i].qty = itm.qty;
+        this.setState(tempObj,()=>{
+            this.evaluateTotalPrice();
+        });
+    }
+    decreaseQty=(e,itm,i)=>{
+        let tempObj = Object.assign({},this.state);
+        itm.qty = itm.qty - 1;
+        if(itm.qty == 0){
+            tempObj.itemList.splice(i,1);
+        }
+        if(!tempObj.itemList.length){
+            this.props.resetCartData();
+        }
+        this.setState(tempObj,()=>{
+            this.evaluateTotalPrice();
```

Comment:
there is no need to setstate twice, first a line no 40 then from inside evaluateTotalPrice function. Instead return new price from evaluateTotalPrice and set both values at once

Reply:
evaluateTotalPrice function is a generic function that sets the total cart value. It is used in increase qty, decrease qty, and on componentDidMount..So, I used it in that way to evaluate the final cart value in the footer.